src/response.rs

```diff
@@ -469,7 +469,12 @@ fn parse_status_line(line: &str) -> Result<(ResponseStatusIndex, u16), Error> {
     }
     // https://tools.ietf.org/html/rfc7230#section-3.1.2
     //      status-line = HTTP-version SP status-code SP reason-phrase CRLF
-    let split: Vec<&str> = line.splitn(3, ' ').collect();
+    let mut split: Vec<&str> = line.splitn(3, ' ').collect();
+    if split.len() == 2 {
+        // As a special case, we are lenient parsing lines without reason phrase.
```

Comment:
Minor tweak:

```suggestion
        // As a special case, we are lenient parsing lines without a space after the code.
```

(because the reason-phrase itself is indeed optional)